backend/Repositories/UsersRepo.cs

```diff
@@ -84,5 +87,30 @@ public User UpdateRole(int id, UpdateUserRoleRequest update)
 
             return user;
         }
+
+        public List<LeaderboardEntry> GetLeaderboard()
+        {
+            var leaderboard = 
+            _context.Sightings
+            .GroupBy(sighting => sighting.CreatedByUserId)
+            .Select(sighting => new {count = sighting.Count(), userId = sighting.Key})
+            .OrderByDescending(g => g.count)
+            .Join(_context.Users, sighting => sighting.userId, user => user.Id, (sighting, user) => 
+            new{username = user.Username, count = sighting.count});
```

Comment:
I think we want some different indentation here

Reply:
Maybe something like
```suggestion
            return _context.Sightings
                .GroupBy(sighting => sighting.CreatedByUserId)
                .Select(sighting => new {count = sighting.Count(), userId = sighting.Key})
                .OrderByDescending(g => g.count)
                .Join(
                    _context.Users, 
                    sighting => sighting.userId, 
                    user => user.Id, 
                    (sighting, user) => new LeaderboardEntry { 
                        Username = user.Username, 
                        Count = sighting.count 
                    }
                ).ToList();
```

This way we can also remove quite a lot of the code below as we are returning a list of LeaderboardEntries

Reply:
Refactored as suggested